api/v1beta1/aerospikebackup_webhook.go

```diff
@@ -0,0 +1,364 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/yaml"
+
+	"github.com/abhishekdwivedi3060/aerospike-backup-service/pkg/model"
+	asdbv1 "github.com/aerospike/aerospike-kubernetes-operator/api/v1"
+	"github.com/aerospike/aerospike-kubernetes-operator/controllers/common"
+)
+
+// log is for logging in this package.
+var aerospikeBackupLog = logf.Log.WithName("aerospikebackup-resource")
+
+func (r *AerospikeBackup) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+var _ webhook.Defaulter = &AerospikeBackup{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *AerospikeBackup) Default() {
+	aerospikeBackupLog.Info("default", "name", r.Name)
+}
+
+//nolint:lll // for readability
+//+kubebuilder:webhook:path=/validate-asdb-aerospike-com-v1beta1-aerospikebackup,mutating=false,failurePolicy=fail,sideEffects=None,groups=asdb.aerospike.com,resources=aerospikebackups,verbs=create;update,versions=v1beta1,name=vaerospikebackup.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &AerospikeBackup{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AerospikeBackup) ValidateCreate() (admission.Warnings, error) {
+	aerospikeBackupLog.Info("validate create", "name", r.Name)
+
+	if len(r.Spec.OnDemandBackups) != 0 && r.Spec.Config.Raw != nil {
+		return nil, fmt.Errorf("onDemand and backup config cannot be specified together while creating backup")
+	}
+
+	if err := r.validateBackupConfig(); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *AerospikeBackup) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	aerospikeBackupLog.Info("validate update", "name", r.Name)
+
+	oldObj := old.(*AerospikeBackup)
+
+	if !reflect.DeepEqual(r.Spec.BackupService, oldObj.Spec.BackupService) {
+		return nil, fmt.Errorf("backup service cannot be updated")
+	}
+
+	if err := r.validateBackupConfig(); err != nil {
+		return nil, err
+	}
+
+	if err := r.validateAerospikeClusterUpdate(oldObj); err != nil {
+		return nil, err
+	}
+
+	if err := r.validateOnDemandBackupsUpdate(oldObj); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *AerospikeBackup) ValidateDelete() (admission.Warnings, error) {
+	aerospikeBackupLog.Info("validate delete", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil, nil
+}
+
+func (r *AerospikeBackup) validateBackupConfig() error {
+	var backupSvc AerospikeBackupService
+
+	cl, gErr := getK8sClient()
+	if gErr != nil {
+		return gErr
+	}
+
+	if err := cl.Get(context.TODO(),
```

Comment:
Let's get the backupService object and do unmarshal after line 150. i.e after validating basic backupConfigMap fields.